core/src/main/scala/sttp/tapir/server/interpreter/RequestBody.scala

```diff
@@ -1,10 +1,19 @@
 package sttp.tapir.server.interpreter
 
 import sttp.capabilities.Streams
-import sttp.tapir.RawBodyType
+import sttp.model.Part
+import sttp.tapir.model.SttpFile
+import sttp.tapir.{RawBodyType, RawPart}
 
 trait RequestBody[F[_], S] {
   val streams: Streams[S]
-  def toRaw[R](bodyType: RawBodyType[R]): F[R]
+  def toRaw[R](bodyType: RawBodyType[R]): F[RawValue[R]]
   def toStream(): streams.BinaryStream
 }
+
+case class RawValue[R](value: R, createdFiles: Seq[SttpFile] = Nil)
+
+object RawValue {
+  def fromParts(parts: Seq[RawPart]): RawValue[Seq[RawPart]] =
+    RawValue(parts, parts collect { case _ @Part(_, f, _, _) if f.isInstanceOf[SttpFile] => f.asInstanceOf[SttpFile] })
```

Comment:
[minor] I think `case _ @ Part(_, f: SttpFile, _, _) => ` would maybe be more succint?